src/unix/core.c

```diff
@@ -1215,6 +1223,13 @@ int uv__getpwuid_r(uv_passwd_t* pwd) {
   pwd->shell = pwd->homedir + homedir_size;
   memcpy(pwd->shell, pw.pw_shell, shell_size);
 
+  /* Copy the gecos field */
+  if (gecos_size > 0) {
+    pwd->gecos = pwd->shell + shell_size;
+    memcpy(pwd->gecos, pw.pw_gecos, gecos_size);
+  } else
+    pwd->gecos = "";
```

Comment:
You could have a `static char empty_string[] = ""`, that would pacify the compiler, but I'd probably opt for allocating that extra byte.